Generic_AdminActions_Default.php

```diff
@@ -273,6 +273,36 @@ private function _w3tc_save_options_process() {
 			}
 		}
 
+		/**
+		 * Page Cache tab
+		 */
+		if ( 'w3tc_pgcache' === $this->_page ) {
+			// Textarea settings
+			$settings = array(
+				'pgcache.mirrors.home_urls',
+				'pgcache.purge.pages',
+				'pgcache.accept.qs',
+				'pgcache.reject.ua',
+				'pgcache.reject.cookie',
+				'pgcache.reject.uri',
+				'pgcache.reject.categories',
+				'pgcache.reject.tags',
+				'pgcache.reject.authors',
+				'pgcache.reject.custom',
+				'pgcache.accept.files',
+				'pgcache.accept.uri',
+				'pgcache.cache.headers',
+			);
+			// Remove empty lines, trim values, and sort.
+			foreach ( $settings as $setting ) {
+				if ( ! empty( $config->get_array( $setting ) ) ) {
+					$values = array_map( 'trim', array_filter( $config->get_array( $setting ) ) );
```

Comment:
I don't see a callback function in array_filter().  Maybe you were trying to use it like this example:
https://www.php.net/manual/en/function.array-filter.php#114931
```
$value = $config->get_array( $setting );

if ( ! empty( $value ) ) {
    $array = array_filter( array_map( 'trim', $value ), 'strlen' );
```
That will trim all array values and then remove empty lines.